File: shoring/compat/src/main/java/com/panopset/compat/Numberop.java
package com.panopset.compat;

public class Numberop {

	public static int parseInt(String value) {
		if (!Stringop.isPopulated(value)) {
			return 0;
		} else {
			String str = value.replace(",", "");
			try {
				return Integer.parseInt(str);
			} catch (NumberFormatException var3) {
				Logop.dspmsg(value);
				Logop.handle(var3);
				return 0;
			}
		}
	}

	public static int parse(String value, Integer base, Integer defaultValue) {
		if (!Stringop.isPopulated(value)) {
			return defaultValue;
		} else {
			String str = value.replace(",", "");
			try {
				return Integer.parseInt(str, base);
			} catch (NumberFormatException var4) {
				Logop.handle(var4);
				return defaultValue;
			}
		}
	}

	public static int parse(String value, Integer base) {
		if (!Stringop.isPopulated(value)) {
			return -1;
		} else {
			String str = value.replace(",", "");

			try {
				return Integer.parseInt(str, base);
			} catch (NumberFormatException var4) {
				Logop.handle(var4);
				return -1;
			}
		}
	}

	public static boolean isNumber(String value) {
		if (value == null) {
			return false;
		}
		return value.matches("[0-9]*");
	}

	public static boolean isInteger(String value) {
		if (value == null) {
			return false;
		}
		return value.matches("-?\\d+");
	}
}

File: code/shoring/compat/src/main/kotlin/com/panopset/compat/Numberop.kt
package com.panopset.compat

import com.panopset.compat.Stringop.isPopulated

object Numberop {
    fun parseInt(value: String): Int {
        if (!isPopulated(value)) {
            return 0
        } else {
            val str = value.replace(",", "")
            try {
                return str.toInt()
            } catch (var3: NumberFormatException) {
                Logz.errorMsg(value, var3)
                return 0
            }
        }
    }

    fun parse(value: String, base: Int, defaultValue: Int): Int {
        if (!isPopulated(value)) {
            return defaultValue
        } else {
            val str = value.replace(",", "")
            try {
                return str.toInt(base)
            } catch (var4: NumberFormatException) {
                Logz.errorEx(var4)
                return defaultValue
            }
        }
    }

    fun parse(value: String, base: Int?): Int {
        if (!isPopulated(value)) {
            return -1
        } else {
            val str = value.replace(",", "")

            try {
                return str.toInt(base!!)
            } catch (var4: NumberFormatException) {
                Logz.errorEx(var4)
                return -1
            }
        }
    }

    fun isNumber(value: String?): Boolean {
        if (value == null) {
            return false
        }
        return value.matches("[0-9]*".toRegex())
    }

    fun isInteger(value: String?): Boolean {
        if (value == null) {
            return false
        }
        return value.matches("-?\\d+".toRegex())
    }
}
